homeassistant/components/zeroconf/__init__.py

```diff
@@ -145,16 +145,22 @@ def addresses(self) -> list[str]:
 @bind_hass
 async def async_get_instance(hass: HomeAssistant) -> HaZeroconf:
     """Zeroconf instance to be shared with other integrations that use it."""
-    return cast(HaZeroconf, (await _async_get_instance(hass)).zeroconf)
+    return cast(HaZeroconf, (_async_get_instance(hass)).zeroconf)
 
 
 @bind_hass
 async def async_get_async_instance(hass: HomeAssistant) -> HaAsyncZeroconf:
     """Zeroconf instance to be shared with other integrations that use it."""
-    return await _async_get_instance(hass)
+    return _async_get_instance(hass)
 
 
-async def _async_get_instance(hass: HomeAssistant, **zcargs: Any) -> HaAsyncZeroconf:
+@callback
+def async_async_instance(hass: HomeAssistant) -> HaAsyncZeroconf:
```

Comment:
that is a terrible name 😄 

and it's also a bit confusing with `async_get_async_instance`. Can we add in the doc string when to use which ?

Reply:
Yeah it's bad. Need to come up with something better...

Reply:
async_get_async_instance_callback is probably a bit less horrid

Reply:
async_get_async_zeroconf seems better